eis_toolkit/prediction/cnn_classification_and_probability.py

```diff
@@ -0,0 +1,353 @@
+
+import numpy as np
+import tensorflow as tf
+from beartype import beartype
+from keras import Model
+from numpy import ndarray
+from sklearn.utils.class_weight import compute_sample_weight
+
+from eis_toolkit.exceptions import InvalidArgumentTypeException, InvalidParameterValueException
+
+
+@beartype
+def _create_an_instance_of_cnn(
+    input_shape_for_cnn: Union[tuple[int, int, int], tuple[int, int]],
+    convolution_kernel_size: tuple[int, int],
+    conv_list: list[int],
+    neuron_list: list[int],
+    pool_size: int = 2,
+    dropout_rate: Union[None, float] = None,
+    last_activation: Literal["softmax", "sigmoid"] = "softmax",
+    regularization: Union[tf.keras.regularizers.L1, tf.keras.regularizers.L2, tf.keras.regularizers.L1L2, None] = None,
+    data_augmentation: bool = False,
+    optimizer: str = "Adam",
+    loss=Union[tf.keras.losses.BinaryCrossentropy(), tf.keras.losses.CategoricalCrossentropy()],
+    output_units=2,
+) -> tf.keras.Model:
+    """
+     Do an instance of the CNN. Just the body of the CNN.
+
+    Args::
+       input_shape_for_cnn: Shape of the input. How big the input should be. It is possible to build an input:
+                         - windows: (h, w, c) windows height, windows width and channel or windows dimension.
+                         - point (p, c) point value and dimension of the point.
+       convolution_kernel_size: Size of the kernel (usually is the last number of the input tuple). Usually is the "c"
+                                of the input_shape_for_cnn.
+       pool_size: Size of the pooling layer (How much you want to reduce the dimension of the data).
+       conv_list: list of unit for the conv layers. Here the length of the list is the number of convolution layers.
+       regularization: Regularization of each  layer. None if you do not want it:
+                     - None if you prefer to avoid regularization.
+                     - L1 for L1 regularization.
+                     - L2 for L2 regularization.
+                     - L1L2 for L1 L2 regularization.
+       data_augmentation: If you want data augmentation or not (Random rotation is implemented).
+       optimizer: Select one optimizer for the CNN. The default value is Adam.
+       loss: The loss function used to measure how good the CNN is performing.
+       neuron_list: List of unit or neuron used to build the network final part of the network. the length of the list
+                    is the number of fully connected layers.
+       dropout_rate: Float number that help avoiding over-fitting. It just randomly drops samples.
+       output_units: number of output classes.
+       last_activation: usually you should use softmax or sigmoid.
+
+    Return:
+        model: the model that has been created.
+
+    Raises:
+        InvalidParameterValueException: Raised when the input is not valid.
+        InvalidParameterValueException: Raised when parameters like dropout is invalid.
+    """
```

Comment:
There is a lot of formatting issues here. Please check:
- indentation
- how multiline parameter descriptions are formatted in other docstring (remove the dashes and the new lines start more to left)
- Capitalization missing from some lines
- In the return statement, remove the "model". Simply describe what is returned
- There is two `:` after `Args`, remove one

Reply:
(About indentation) Sorry man for that is my pycharm is crazy :-(